proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+
+    public boolean isPermissionParameterAllowed(FunctionID rpcName, String parameter){}
+
+    public void dispose(){}
+
+    // Determine if a group of permissions are allowed for the current HMI level
+    public @PermissionGroupStatus int getGroupStatusOfPermissions(List<PermissionElement> permissionElements){}
+
+    // Retrieve a map with keys that are the passed in RPC names specifying if that RPC
+    // and its parameter permissions are currently allowed for the current HMI level
+    public Map <FunctionID, PermissionStatus> getStatusOfPermissions(List<PermissionElement> permissionElements){}
+
+    public UUID addListener(List<PermissionElement> permissionElements, @PermissionGroupType int groupType, OnPermissionChangeListener listener){}
+
+    // Removes specific listener
+    public void removeListener(UUID listenerId){}
+
+    public interface OnPermissionChangeListener {
+        void onPermissionsChange(Map <FunctionID, PermissionStatus> allowedPermissions, @PermissionGroupStatus int permissionGroupStatus);
+    }
+
+}
+```
+
+The `PermissionElement` class holds an RPC name and its parameter permissions.
+
+```java
+public class PermissionElement {
+    PermissionElement(FunctionID rpcName, List<String> parameters){}
+
+    public FunctionID getRpcName() {}
+
+    public void setRpcName(FunctionID rpcName) {}
+
+    public List<String> getParameters() {}
+
+    public void setParameters(List<String> parameters) {}
+}
+```
+
+The `PermissionStatus` class is a small class that represents whether an individual RPC and its parameters are allowed or not.
+
+```java
+class PermissionStatus {
+    public PermissionStatus(FunctionID rpcName, boolean isRPCAllowed, Map<String, Boolean> allowedParameters) {}
+
+    public FunctionID getRpcName() {}
+
+    public boolean getIsRPCAllowed() {}
+
+    public Map<String, Boolean> getAllowedParameters() {}
+}
+```
+
+The `PermissionFilter` holds all the required information for a specific listener.
+
+```java
+class PermissionFilter {
+    PermissionFilter(UUID identifier, List<PermissionElement> permissionElements, int groupType, PermissionManager.OnPermissionChangeListener listener) {}
+
+    UUID getIdentifier() {}
+
+    List<PermissionElement> getPermissionElements() {}
+
+    int getGroupType() {}
+
+    PermissionManager.OnPermissionChangeListener getListener() {}
+}
+```
+
+### Other Additions
+
+#### SDLProxyBridge
+
+The SDLProxyBridge has been implemented to allow the library to bridge from old API's to new. The `SdlManager` will hold an instantiate of this class that is passed to its proxy object. Using the `SdlProxyBase` as our main interface into the old APIs with this bridge the manager API change can be added with only a minor version change. Proxy bridge allows the managers to listen for specific RPCs without having to individually implement `IProxyListenerBase` themselves and having to override all of the response methods like what is currently required.
```

Comment:
Please update "APIs" (removing apostrophe) in the first sentence and add a comma to the second sentence so that it reads more clearly: "...with this bridge, the manager API change..."